### PR TITLE
[fix](Nereids): round microsecond when specify scale of microsecond

### PR DESCRIPTION
**Pull request by jackwener**
## Proposed changes

select cast('0001-01-02 00:00:00.514514' as datetime(3));

should handle it to be: '0001-01-02 00:00:00.515000'

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by jackwener** [comment]
run buildall

**Comment by jackwener** [comment]
run buildall

**Comment by jackwener** [comment]
run buildall

**Comment by jackwener** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/literal/DateTimeV2Literal.java [hunk 1]
```diff
@@ -48,9 +51,30 @@ public DateTimeV2Literal(long year, long month, long day, long hour, long minute
...
+    public DateTimeV2Literal(DateTimeV2Type dateType,
             long year, long month, long day, long hour, long minute, long second, long microSecond) {
-        super(dataType, year, month, day, hour, minute, second, microSecond);
+        super(dateType, year, month, day, hour, minute, second, microSecond);
+        roundMicroSecond(dateType.getScale());
+    }
+
+    private void roundMicroSecond(int scale) {
+        Preconditions.checkArgument(scale >= 0 && scale <= DateTimeV2Type.MAX_SCALE,
+                "invalid datetime v2 scale: " + scale);
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/literal/DateTimeV2Literal.java:62]: it is better to use msg format template
```suggestion
        Preconditions.checkArgument(scale >= 0 && scale <= DateTimeV2Type.MAX_SCALE,
                "invalid datetime v2 scale: {}", scale);
```

**Review by starocean999 — APPROVED** [review]

**Pull request merged**